src/index.js

```diff
@@ -616,58 +616,120 @@ class Service extends AdapterService {
       });
   }
 
-  /**
-   * `create` service function for Objection.
-   * @param {object} data
-   * @param {object} params
-   */
-  async _create (data, params) {
-    const transaction = await this._createTransaction(params);
-    const create = (data, params) => {
-      const q = this._createQuery(params);
-      const allowedUpsert = this.mergeRelations(this.allowedUpsert, params.mergeAllowUpsert);
-      const allowedInsert = this.mergeRelations(this.allowedInsert, params.mergeAllowInsert);
-
-      if (this.createUseUpsertGraph) {
-        if (allowedUpsert) {
-          q.allowGraph(allowedUpsert);
-        }
-        q.upsertGraphAndFetch(data, this.upsertGraphOptions);
-      } else if (allowedInsert) {
-        q.allowGraph(allowedInsert);
-        q.insertGraph(data, this.insertGraphOptions);
-      } else {
-        q.insert(data, this.id);
-      }
-
-      return q
-        .then(row => {
-          if (params.query && params.query.$noSelect) { return data; }
+  _getCreatedRecords (insertResults, inputData, params) {
+    if (params.query && params.query.$noSelect) {
+      return inputData;
+    }
+    if (!Array.isArray(insertResults)) {
+      insertResults = [insertResults];
+    }
 
-          let id;
+    const findQuery = Object.assign({ $and: [] }, params.query);
+    const idsQueries = [];
 
-          if (Array.isArray(this.id)) {
-            id = [];
+    if (Array.isArray(this.id)) {
+      for (const insertResult of insertResults) {
+        const ids = [];
 
-            for (const idKey of this.id) {
-              id.push(row && row[idKey] ? row[idKey] : data[idKey]);
-            }
+        for (const idKey of this.id) {
+          if (idKey in insertResult) {
+            ids.push(insertResult[idKey]);
           } else {
-            id = row && row[this.id] ? row[this.id] : data[this.id];
+            return inputData;
           }
+        }
+        idsQueries.push(this.getIdsQuery(ids));
+      }
+    } else {
+      const ids = [];
+      for (const insertResult of insertResults) {
+        if (!(typeof insertResult === 'object' && insertResult !== null)) {
+          ids.push(insertResult);
+          continue;
+        }
 
-          if (!id || (Array.isArray(id) && !id.length)) { return data; }
+        if (this.id in insertResult) {
+          ids.push(insertResult[this.id]);
+        } else {
+          return inputData;
+        }
+      }
+      idsQueries.push(this.getIdsQuery(null, ids));
+    }
 
-          return this._get(id, params);
-        })
-        .catch(errorHandler);
-    };
+    if (idsQueries.length > 1) {
+      findQuery.$and.push({ $or: idsQueries });
```

Comment:
Because you can't use `$in` operator with a compound PK. The generated SQL should be something like this:

```sql
select `people-rooms`.* from `people-rooms` 
where (`peopleId` = ? and `roomId` = ?) or (`peopleId` = ? and `roomId` = ?) or (`peopleId` = ? and `roomId` = ?)
```

Reply:
If this is the generated query, then it looks good. Did you test it?

Reply:
Yes, this is the generated query, but slightly refactored. I tested it with SQLite, PostgreSQL and MySQL.